### ISSUE TITLE
Can't use --dict-tokenixer bytelevelbpe with Blender 90M?

### ISSUE THREAD

**Issue by aslu98** [facebookresearch/ParlAI#2982]
I'm trying to train a model with command:
```
parlai train_model  -t fromfile:parlaiformat \
       --fromfile_datapath ../parlai_utilities/covidQA_parlaiformat.txt -m transformer/generator \
         --init-model zoo:blender/blender_90M/model --dict-file zoo:blender/blender_90M/model.dict \
         -veps 0.25 --attention-dropout 0.0 -bs 1  --embedding-size 2560 \
          --ffn-size 10240 --variant prelayernorm --n-heads 32 --n-positions 128 \
         --n-encoder-layers 2 --n-decoder-layers 24 \
         --history-add-global-end-token end --delimiter ' ' \
         --dict-tokenizer bytelevelbpe --dropout 0.1 --fp16 True \
         --label-truncate 32 --log_every_n_secs 10 -lr 7e-06 --lr-scheduler reduceonplateau \
         --lr-scheduler-patience 3 --optimizer sgd --relu-dropout 0.0 --activation gelu \
  	 --text-truncate 32 --truncate 32 \
         --warmup_updates 100 --update-freq 2 --gradient-clip 0.1 \
         --model-parallel True --skip-generation True -vp 10 -vmt ppl -vmm min \
         --num-epochs 1.0 --save-after-valid True --model-file covid90M
```

but I get this error:
**`OSError: --bpe-vocab and --bpe-merge are mandatory with --dict-tokenizer bytelevelbpe`**
But when I train it with the same command starting from Blender 3B everything is fine.
I noticed from the fine-tuning section of the Recipes page that 3B and 9B models use `--dict-tokenizer bytelevelbpe` but 90M model uses `--dict-tokenizer bpe`. Why can't I use bytelevelbpe with 90M model? Or, if I need to use `--bpe-vocab` and `--bpe-merge` what value should I give?

Complete Log:
```
Traceback (most recent call last):
  File "/usr/local/bin/parlai", line 33, in <module>
    sys.exit(load_entry_point('parlai', 'console_scripts', 'parlai')())
  File "/parlai/core/script.py", line 272, in superscript_main
    SCRIPT_REGISTRY[cmd].klass._run_from_parser_and_opt(opt, parser)
  File "/parlai/core/script.py", line 88, in _run_from_parser_and_opt
    return script.run()
  File "/parlai/scripts/train_model.py", line 769, in run
    self.train_loop = TrainLoop(self.opt)
  File "/parlai/scripts/train_model.py", line 278, in __init__
    self.agent = create_agent(opt)
  File "/parlai/core/agents.py", line 405, in create_agent
    model = model_class(opt)
  File "/parlai/core/torch_generator_agent.py", line 474, in __init__
    super().__init__(opt, shared)
  File "/parlai/core/torch_agent.py", line 723, in __init__
    self.dict = self.build_dictionary()
  File "/parlai/core/torch_agent.py", line 807, in build_dictionary
    d = self.dictionary_class()(self.opt)
  File "/parlai/core/dict.py", line 305, in __init__
    self.bpe = bpe_factory(opt, shared)
  File "/parlai/utils/bpe.py", line 59, in bpe_factory
    bpe_helper = HuggingFaceBpeHelper(opt, shared)
  File "/parlai/utils/bpe.py", line 734, in __init__
    '--bpe-vocab and --bpe-merge are mandatory with '
```

**Comment by stephenroller** [comment]
The 90M model uses bpe, not bytelevelbpe.

**Comment by aslu98** [comment]
Thanks, I think that it would be useful to specify that you are forced to use bpe with 90M model in the Error.

**Comment by stephenroller** [comment]
Agreed, we need to solve the issue of which opts are allowed to change and which aren't. It's tracked in a wider issue.

Can we close this one?

**Comment by aslu98** [comment]
Sure, I'm sorry, I didn't noticed that. I'm closing this Issue.

**Comment by stephenroller** [comment]
It's a very generic issue; I wouldn't have expected you to notice.

**Comment by aslu98** [comment]
I'm re-opening the issue because now I'm working with `--dict-tokenizer bytelevelbpe` and the 3B model but I still get the error.

After getting a Type Error like the one in #2931, which I solved adding the save function (#2918) as suggested, now I get  `OSError: --bpe-vocab and --bpe-merge are mandatory with --dict-tokenizer bytelevelbpe` again.

If I change `--dict-tokenizer`  to `bpe` it works but my ppl is much worse (from around 16 to around 16000) so I would really like to use `bytelevelbpe`.

**Comment by stephenroller** [comment]
Bytelevelbpe and bpe tokenizers are different and mutually exclusive; the ppl change is expected.

You do need to give the args when using bytelevelbpe.

Will look into this more later.
